storm-server/src/test/java/org/apache/storm/scheduler/resource/TestUtilsForResourceAwareScheduler.java

```diff
@@ -83,6 +88,10 @@ public void addSelfTo(Map<String, Map<String, Number>> fullPool) {
         }
     }
 
+    public static TestUserResources userRes(String name, Map<String, Double> resources) {
+        return new TestUserResources(name, resources);
```

Comment:
This reminds me that we don't have a unit test to test the case where users have the generic resource guarantee. But this can be a follow up work

Reply:
Addressed all comments, thanks for the prompt review. As for this, yes we should have some tests on User class. Currently we only have a testResourcePoolUtilization. When I was coding I also extended the usage of pool utilization related functions to include generic resources as well. But after I realized that there is nowhere we are using it. I will filed another PR for removing them.

Reply:
Right. But what I meant is unit tests for GenericResourceAwareSchedulingPriorityStrategy when users do have the generic resource guarantee